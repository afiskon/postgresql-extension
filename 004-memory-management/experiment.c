#include <postgres.h>
#include <utils/builtins.h>

PG_MODULE_MAGIC;
PG_FUNCTION_INFO_V1(experiment_palloc);

Datum
experiment_palloc(PG_FUNCTION_ARGS)
{
	char *buffcopy, *fmtstr;
	char *mybuff = (char*)palloc(128); /* or palloc0() */
	snprintf(mybuff, 128, "test data");
	elog(NOTICE, "mybuff after palloc() = %s", mybuff);

	mybuff = repalloc(mybuff, 256);
	elog(NOTICE, "mybuff after repalloc() = %s", mybuff);

	buffcopy = pstrdup(mybuff);
	pfree(mybuff);

	elog(NOTICE, "byffcopy = %s", buffcopy);

	fmtstr = psprintf("This is %s example", "psprintf()");
	elog(NOTICE, "fmtstr = %s", fmtstr);


	PG_RETURN_VOID();
}
