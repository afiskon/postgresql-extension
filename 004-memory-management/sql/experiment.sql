CREATE EXTENSION experiment;

SELECT experiment_palloc();

DROP EXTENSION experiment;