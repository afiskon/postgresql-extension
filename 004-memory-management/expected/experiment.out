CREATE EXTENSION experiment;
SELECT experiment_palloc();
NOTICE:  mybuff after palloc() = test data
NOTICE:  mybuff after repalloc() = test data
NOTICE:  byffcopy = test data
NOTICE:  fmtstr = This is psprintf() example
 experiment_palloc 
-------------------
 
(1 row)

DROP EXTENSION experiment;
